Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py

```diff
@@ -90,8 +90,16 @@ def Runner(env):
         if path_to_os is not None:
             # Potentially dealing with big daddy, give it more juice...
             args += " -m 8192"
-            #args += " -hda \"" + path_to_os + "\""
-            args += " -drive format=raw,index=0,media=disk,file=\"" + path_to_os + "\""
+
+            storage_rule = {

```

Comment:
For new code additions, I suggest using f-strings so the substitution is less verbose and `pathlib` to get the extension via `.suffix` which is more straightforward to read.

Something like:

```suggestion
            file_extension = Path(path_to_os).suffix.lower().replace('"', '')

            storage_rule = {
                ".vhd": f" -drive format=raw,index=0,media=disk,file=\"{path_to_os}\"",
                ".qcow2": f" -hda \"{path_to_os}\""
            }.get(file_extension, None)

```

Makes it easier to read.

Reply:
Agreed, I only left it this way to keep compatibility with python 3.6 if that's not a concern I'm happy to make this change

Reply:
We're past that point now in a number of scripts.

Reply:
pytools provides support for 3.9, 3.10, and 3.11 so any pytool scripts need only support those versions.